logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -217,6 +218,12 @@ def cancel(event)
         # @cancelled[event] = true
       end
 
+      def collect
```

Comment:
Similar comment to wrapped_acked_queue: I recommend removing `collect` method and adding `include Enumerable` which will add both `collect` and `to_a` (and we should use `to_a` if we wish an array of elements)

Reply:
@andrewvc hmm -1 :D this really isn't the kind of method you'd expect behind `to_a` because we're also filtering out and joining two internal collections here (there's lots of magic in `each` imo). Imo (in FP), `collect` is precisely the name you'd put on that kind of operation?

Reply:
hmm I don't agree, but I also don't think it's important enough to argue. Renaming but still:
* you have argument free in at least `collect` in Spark + Scala (in a sense also in MapReduce) it's 
* `to_a` is a nasty name to bring into you Java APIs
*  the fact that there is magic like `uncancellel` going on behind the scenes makes this kinda meh just present as `to_a` still push incoming :D